### DISCUSSION TITLE
远程同步功能未按照设定生效

### DISCUSSION THREAD

**Discussion by chenkuicn** [halo-dev/halo#10007] [General]
![Image](https://github.com/user-attachments/assets/12a0d7f4-7181-4565-bd61-aaf484deb2ff)
![Image](https://github.com/user-attachments/assets/43f57eee-e0ef-45c0-bc87-2e4303b0a5ab)
![Image](https://github.com/user-attachments/assets/c49095fa-e89e-49d6-b595-5eacb2f6ffb6)
![Image](https://github.com/user-attachments/assets/47f38630-5e88-4546-afa6-7e26743159d4)
![Image](https://github.com/user-attachments/assets/4e480222-6f9e-4e2b-a972-c91b21c29464)

本地的备份会按照设定只保留三份，但是远程同步会出现四份或者五份（没测试更多，四份五份的时候手动删除多余的同步任务）。并且远程同步删除多余的的备份时会同步失败，不确定是不是判定本地已经没有这个备份了（本地备份正常），所以远程同步失败。
另外最后一张图片提示有点歧义，手动删除远程同步任务时会同时删除S3桶内文件。但是不确定有没有删除本地备份，没去测试。所以如果不删除本地备份数据的话，这句话应该改成”但不会删除本地备份数据“。

**Comment by JohnNiang** [comment]
Hi @chenkuicn , thank you for reaching out here!

> 当前问题主要和备份插件有关系。

1. 因为备份插件检测成功记录数的时机是在执行同步任务之前，当执行完同步任务之后，这里会比预设的“保留成功记录数”多出一份。或许后续可以考虑在执行同步任务之后再执行清理操作。

2. 删除同步任务的提示语确实可以优化一下。目前删除同步任务只会删除当前同步任务以及已经同步好的远程备份文件，确实不会删除当前系统中的本地备份文件。
